### PR TITLE
feat:export additional graphql types

### LINKED ISSUE #2557: Improving GraphQL extensions and documentation
# Documentation Issue

I think documentation for extending GraphQL is lacking when it comes to covering a full customisation a user may realistically need. I covered some of this in my tutorial here https://nouance.io/articles/how-to-extend-graphql-in-payload-with-custom-queries-mutations-and-re-usable-types but I want to bring some of those examples and information into the official docs.

Parts that are confusing:
- Available types for my resolvers, return types and so on
- Best practices for code structure
- Explaining how to type and which graphql packages are used and where to dig deeper if you want more power (the source documentation)
- Things to consider when using the LocalAPI within a graphql resolver
  - eg. the issue I struggled with when accidentally setting (or emitting) the `depth` argument 

## Additional Details

Before updating the docs I actually want to expose a few types and utilities from Payload's core.

- [ ] Expose the `buildPaginatedListType` function
- [ ] Expose the type for a GraphQL query or mutation from the internal type `(GraphQL: typeof ImportedGraphQL, payload: Payload) => Record<string, unknown>`
- [ ] Expose Payload's GraphQL type too, currently you have to kind of do this `import type { default as ImportedGraphQL } from 'graphql'`
- [ ] Attach the collection's already built paginatedType to the `.graphQL.` object so that it can be properly re-used in custom queries, making the re-use of fragments in the frontend a lot easier

The goals here are to reduce friction in working with and extending GraphQL. Happy to work on this and add a PR for the code and documentation updates if there's a greenlight from the Payload team

### PR DESCRIPTION
**Pull request by paulpopus**
## Description

Closes https://github.com/payloadcms/payload/issues/2557 , I will update the documentation around GraphQL after this PR in case there are any further changes needed in this code.

This PR exports:
- buildPaginatedListType
- the internally used GraphQL library and type to make sure there's no conflicts (happy to revert on this one but I thought it might make the DX easier)
- the mutations and queries type into a re-usable type `GraphQLExtension` in the config

And it attaches the created `buildPaginatedType` from collections to the available types for re-use on `payload.collections.graphQL.`

- [x] I have read and understand the CONTRIBUTING.md document in this repository

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update (will come in a PR after this one)

### CODE REVIEW

File: graphql.d.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export { default as buildPaginatedListType } from "./dist/graphql/schema/buildPaginatedListType";
+export {default as GraphQL} from 'graphql';
```
Comment (denolfe) [inline graphql.d.ts:2]: Let's clean up the spacing and quote type here.

File: src/collections/graphql/init.ts [hunk 2]
```diff
@@ -95,10 +95,15 @@ function initCollectionsGraphQL(payload: Payload): void {
...
+    collection.graphQL.paginatedType = buildPaginatedListType(
+      pluralName,
+      collection.graphQL.type
+    );
+
     collection.graphQL.whereInputType = buildWhereInputType(
       singularName,
       whereInputFields,
-      singularName,
+      singularName
```
Comment (denolfe) [inline src/collections/graphql/init.ts:106]: Add back comma

File: graphql.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+exports.buildPaginatedListType =
+  require("./dist/graphql/schema/buildPaginatedListType").default;
+
+exports.GraphQL = require("graphql");
```
Comment (denolfe) [inline graphql.js:4]: Discussed this with the team, we're okay with this 👍

**Review by denolfe — CHANGES_REQUESTED** [review]
Just some formatting tweaks. Other than that looks good. 

Will re-review when docs are added 👍

**Comment by paulpopus** [comment]
@denolfe I've fixed those code style issues. Added documentation, let me know how it looks.

Going through it again, there's probably another round of work to do to make working with GraphQL feel more first class.

**Review by denolfe — APPROVED** [review]

**Pull request merged**